cozmo_soar.py

```diff
@@ -648,6 +654,153 @@ def on_input_phase(self, input_link: sml.Identifier):
+                        tag_list.append(f'tag add {oid} is_visible True')
+                        buff[oid] = deepcopy(val)
+            
+        def get_obj_str(obj, str_type):
+            position = obj.pose.position
+            rotation = obj.pose.rotation
+            pos_arr = [position.x, position.y, position.z]
+            quat = (rotation.q0, rotation.q1, rotation.q2, rotation.q3)
+            oid = obj.object_id
+
+            def quaternion_to_euler(quat):
+                """
+                credit: https://stackoverflow.com/questions/53033620/how-to-convert-euler-angles-to-quaternions-and-get-the-same-euler-angles-back-fr?rq=1
+                """
+                import math
+                (w, x, y, z) = quat
+                t0 = +2.0 * (w * x + y * z)
+                t1 = +1.0 - 2.0 * (x * x + y * y)
+                X = math.atan2(t0, t1)
+
+                t2 = +2.0 * (w * y - z * x)
+                t2 = +1.0 if t2 > +1.0 else t2
+                t2 = -1.0 if t2 < -1.0 else t2
+                Y = math.asin(t2)
+
+                t3 = +2.0 * (w * z + x * y)
+                t4 = +1.0 - 2.0 * (y * y + z * z)
+                Z = math.atan2(t3, t4)
+                return (X, Y, Z)
+
+            eul = quaternion_to_euler(quat)
+
+            if type(obj).__name__ == '_proxy_LightCube':
+                LIGHT_CUBE_LENGTH = 43
+                depth = LIGHT_CUBE_LENGTH
+                width = LIGHT_CUBE_LENGTH
+                height = LIGHT_CUBE_LENGTH
+            
+            else:
+                depth = obj.x_size_mm
+                width = obj.y_size_mm
+                height = obj.z_size_mm
+
+            # generate [GEOMETRY] for svs str by generating vertices from position and vectors of dimensions      
+            def combinations(possibilities):
+                """
+                Takes in array of the form [(tuple), (tuple), (tuple)] and returns all the possible combinations.  You can think of each tuple as possibilities of a value at a certain place (index in top level array).  This returns all possible combinations of those values in each place.
+                """
+                # will be modified during recursion
+                combs = []
+                def comb_helper(arr, i, baton):
+                    """
+                    Modifies combs variable above
+                    """
+
+                    for elt in arr[i]:
+                        baton.append(elt)
+                        if i < len(arr) - 1:
+                            comb_helper(arr, i + 1, baton)
+                        else:
+                            combs.append(copy.deepcopy(baton))
+                        baton.pop()
+                comb_helper(possibilities, 0, [])
+                return combs
+
+            # vectors from center of object to vertices, unrotated
+            vectors = np.transpose(np.array(combinations([(depth/2.0, -depth/2.0), (width/2.0, -width/2.0), (height/2.0, -height/2.0)])))
```

Comment:
Instead of custom function, use built-in:

```suggestion
            raw_vectors = itertools.product((depth/2.0, -depth/2.0), (width/2.0, -width/2.0), (height/2.0, -height/2.0))
            vectors = np.transpose(np.array(raw_vectors))
```